packages/store-sync/src/zustand-query/createStorageAdapter.ts

```diff
@@ -0,0 +1,147 @@
+import { hexToResource, resourceToLabel, spliceHex } from "@latticexyz/common";
+import {
+  KeySchema,
+  decodeKey,
+  decodeValueArgs,
+  encodeValueArgs,
+  getKeySchema,
+  getSchemaTypes,
+  getValueSchema,
+} from "@latticexyz/protocol-parser/internal";
+import { size } from "viem";
+import { isTableRegistrationLog } from "../isTableRegistrationLog";
+import { logToTable } from "../logToTable";
+import { StorageAdapter, StorageAdapterBlock } from "../common";
+import { Store } from "@latticexyz/zustand-query/internal";
+import { debug } from "./debug";
+
+export type CreateStorageAdapterOptions = {
+  store: Store;
+};
+
+export type CreateStorageAdapterResult = {
+  storageAdapter: StorageAdapter;
+};
+
+const emptyValueArgs = { staticData: "0x", encodedLengths: "0x", dynamicData: "0x" } as const;
+
+export function createStorageAdapter({ store }: CreateStorageAdapterOptions): CreateStorageAdapterResult {
+  async function storageAdapter({ logs }: StorageAdapterBlock): Promise<void> {
+    const newTables = logs.filter(isTableRegistrationLog).map(logToTable);
+
+    for (const newTable of newTables) {
+      // TODO: switch this to `newTable.label` once available
+      const existingTable = store.getState().config[newTable.namespace]?.[newTable.name];
+      if (existingTable) {
+        console.warn("table already registered, ignoring", {
+          newTable,
+          existingTable,
+        });
+      } else {
+        // TODO: create util for converting table config
+        store.getState().actions.registerTable({
+          ...newTable,
+          key: newTable.key as string[],
+          label: newTable.name,
+          schema: Object.fromEntries(Object.entries(newTable.schema).map(([field, { type }]) => [field, type])),
+        });
+      }
+    }
+
+    // Convert logs in (partial) table updates
+    for (const log of logs) {
+      const { namespace, name } = hexToResource(log.args.tableId);
+      // TODO: use label once available
+      const tableConfig = store.getState().config[namespace][name];
+      const boundTable = store.getState().actions.getTable({ namespace, label: name });
```

Comment:
do we need to use bound tables here if we have the table config? would it be better to use the regular/non-bound API?